[Ming Sheng] iP

moingshoing opened the pull request:
### Aris

> "I don't know my purpose here." - Jae Crowder on his role with the Bucks

Aris frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- _SUPER_ FAST to use

All you need to do is,

1. download it from [here](https://github.com/moingshoing/ip/releases).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

Please send me money I'm **broke**

Features:

- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

`"Hello world"`

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

moingshoing committed:
lvl 1

moingshoing committed:
customise

moingshoing committed:
a

moingshoing committed:
no message

moingshoing committed:
lvl 2

moingshoing committed:
lvl 3

moingshoing committed:
scuffed lvl 4

moingshoing committed:
ui test for bat updated, ignore sh

moingshoing committed:
lvl 5 basic exceptions handled for now

moingshoing committed:
enum and delete

moingshoing committed:
forgot to change one part of code to accomodate for enum

moingshoing committed:
same mistake as the previous commit

moingshoing committed:
updated ui test

moingshoing committed:
text changes; nth important

moingshoing committed:
Aris.java: add loadTodolist
Deadline class: add new constructor
Todo class: add new constructor
Event class: add new constructor

moingshoing committed:
Level 7

moingshoing committed:
Deadline.java: add LocalDate class to deal with dates

moingshoing committed:
Deadline.java: Update dateFormatter to handle DateTimeParseException within method

moingshoing committed:
Event.java: Add dateFormatter method which uses LocalDate class

moingshoing committed:
Storage.java: Add Storage class
TaskList.java: Add TaskList class

moingshoing committed:
TaskList.java:  Add constructor for TaskList
Aris.java: Change code to utilise TaskList methods

moingshoing committed:
Storage.java: Add Storage class which handles loading and saving of files
TaskList.java: Add toFile() method which converts list to file format

moingshoing committed:
Parser.java: Add Parser class to parse user input
Aris.java: Utililse Parser class to parse user input

moingshoing committed:
Divide classes into packages

moingshoing committed:
Merge branch 'master' into add-gradle-support

moingshoing committed:
bug fix: replace emoticons with ASCII compatible characters

moingshoing committed:
ArisTest.java: Add test class

moingshoing committed:
Merge branch 'add-gradle-support'

moingshoing committed:
ArisTest.java: Add two JUnit tests

moingshoing committed:
Merge branch 'master' into branch-A-JUnit

moingshoing committed:
ArisTest.java: Edit JUnit tests

moingshoing committed:
build.gradle: Edit jar file name

moingshoing committed:
Add JavaDoc comments to code

moingshoing committed:
Aris.java: Remove indentation for case clauses and add explicit //Fallthrough comment

moingshoing committed:
TaskList.java: Add findTasks method to look up tasks with same keyword
Task.java: Add matchesKeyword method to check if the task contains the keyword
Command.java: Add FIND
Aris.java: Add new case where user inputs the find command

moingshoing committed:
Ui.java: Change greet message

moingshoing committed:
Merge branch 'branch-A-CodingStandard'

moingshoing committed:
Merge branch 'branch-A-JavaDoc'

moingshoing committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	data/Aris.txt
#	src/main/java/aris/Aris.java

moingshoing committed:
bug fix: remove jargon in code

moingshoing committed:
Storage.java: Add if statements to handle corrupted file error

wallacepck commented on src/main/java/aris/Aris.java:
```diff
@@ -0,0 +1,125 @@
+        } catch(FileNotFoundException e) {
+            arisUi.format("No file found ¯\\_(._.)_/¯");
+        } catch(IllegalArgumentException e) {
+            arisUi.format("File corrupted ¯\\_(._.)_/¯");
```
Should the program continue to run after knowing its savedata is corrupted?

wallacepck commented on src/main/java/aris/list/TaskList.java:
```diff
@@ -0,0 +1,132 @@
+        } catch (IllegalArgumentException e) {
+            return "Number is out of range ¯\\_(._.)_/¯";
+        }
+    }
```
These two functions are almost the same, would it better to combine them into one?

wallacepck commented on src/main/java/aris/ui/Ui.java:
```diff
@@ -0,0 +1,23 @@
+    private String hline = "____________________________________________________________\n";
+    private String lastMessage = "";
+
+    public void format(String s) {
```
```suggestion
    public void printFormattedLine(String s) {
```
Perhaps a name that better describes its behaviour?

wallacepck commented on src/main/java/aris/list/TaskList.java:
```diff
@@ -0,0 +1,132 @@
+        } catch (IllegalArgumentException e) {
+            return "Number is out of range ¯\\_(._.)_/¯";
+        }
+    }
```
I believe combining these same results into an exception or a constant would improve maintainability

wallacepck commented on src/main/java/aris/list/TaskList.java:
```diff
@@ -0,0 +1,132 @@
+        return task.addTask(list);
+    }
+
+    public String findTask(String keyword) {
```
The parameter is named `keyword` but in use there may be more than one grammatical word, perhaps a different name like `keyphrase` that alludes to this would be better?

wallacepck commented on src/main/java/aris/task/Deadline.java:
```diff
@@ -0,0 +1,57 @@
+ * Represents a task with a deadline.
+ */
+public class Deadline extends Task {
+    protected String deadline;
+    protected String deadlineString;
```
I can not tell what the difference between `deadline` and `deadlineString` is at a glance, perhaps a clearer name for both would help?

xinnnyeee replied:
I agree! or perhaps extract it to a method toString(), which is what a lot of people do conventionally. perhaps it helps w people's understanding! ^_^

wallacepck commented on src/main/java/aris/task/Event.java:
```diff
@@ -0,0 +1,62 @@
+public class Event extends Task {
+    protected String from;
+    protected String to;
+    protected String periodString;
```
Instead of using the ambiguous suffix "String", perhaps naming this `formattedPeriod` or a similarly descriptive name would be clearer?

wallacepck commented on src/main/java/aris/task/Task.java:
```diff
@@ -0,0 +1,74 @@
+        // del task from duke.txt
+        list.remove(index - 1);
+        return String.format("Okei this task is deleted:\n%s\nYou now have %d task(s) in the list", status(), list.size());
+    }
```
This violates encapsulation, is there a reason for doing this?

wallacepck commented on src/main/java/aris/task/Task.java:
```diff
@@ -0,0 +1,74 @@
+     * Formats the task for file storage.
+     * @return A formatted string representing the task for storage.
+     */
+    public String fileFormat() {
```
I don't like how the term "file format" is used here, perhaps it would be better to name this `getSerialisedFormat()`?

xinnnyeee replied:
i agree that method should be phrased as actions! maybe getListFormat()

wallacepck commented on src/main/java/aris/task/Task.java:
```diff
@@ -0,0 +1,74 @@
+/**
+ * Represents a general task with basic functionalities.
+ */
+public class Task {
+    protected String description;
+    protected boolean isDone;
```
Is there a reason for `Task` not being abstract?

Additionally, the fields `isDone` and `description` are constructed in subclasses instead of the smallest scope here, would it be better to delegate it to `Task` instead?

wallacepck commented on src/test/java/aris/ArisTest.java:
Perhaps using the given test method format `featureUnderTest_testScenario_expectedBehavior()` would be more readable?

xinnnyeee replied:
that's true, I should work on changing my own test method names too HAHHA

wallacepck commented on data/Aris.txt:
Perhaps add the data folder to gitignore? Especially for release builds, it would be confusing for end-users to see data already in the program unless it is explicitly labelled as an example.

wallacepck commented on src/main/java/aris/list/TaskList.java:
```diff
@@ -0,0 +1,132 @@
+     * Converts the list of tasks into a file-friendly format.
+     * @return A formatted string representing the tasks for storage.
+     */
+    public String toFile() {
```
Perhaps this should be named `toSerialisedFormat` as this function does not actually involve any files.

wallacepck commented on src/main/java/aris/list/TaskList.java:
```diff
@@ -0,0 +1,132 @@
+     * @param index The index to check.
+     * @throws IllegalArgumentException If the index is out of bounds.
+     */
+    public void isIndexValid(int index) throws IllegalArgumentException {
```
Should this return a boolean instead? I would be quite saddened if a function named `is...` immediately throws an exception.

wallacepck commented on src/main/java/aris/storage/Storage.java:
```diff
@@ -0,0 +1,98 @@
+     * @param list The TaskList to load tasks into.
+     * @throws FileNotFoundException If the file does not exist.
+     */
+    public void loadFile(TaskList list) throws FileNotFoundException, IllegalArgumentException {
```
What if an `IOException` is thrown when reading the file? I would prefer this to be documented or caught by the code itself.

wallacepck commented on src/main/java/aris/storage/Storage.java:
```diff
@@ -0,0 +1,98 @@
+            }
+
+            String taskType = parts[0];
+            int isDone = Integer.parseInt(parts[1]);
```
Would it be better for this to be further converted to a boolean? That would prevent any potential bugs with unexpected values and make the code easier to analyse.

wallacepck commented on src/test/java/aris/ArisTest.java:
```diff
@@ -0,0 +1,72 @@
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    public void testIndexOutOfRange() {
+        String actual = list.markTaskDone(4);
+        String expected = "Number is out of range ¯\\_(._.)_/¯";
+    }
```
Is the `assertEquals` missing here?

wallacepck commented on src/main/java/aris/task/Deadline.java:
```diff
@@ -0,0 +1,57 @@
+     * Constructs a Deadline task from a description string.
+     * @param description The task description with deadline.
+     */
+    public Deadline(String description) {
```
I would not use the same name `description` for this parameter, as that implies the given argument is the same description in the `Task`.

wallacepck reviewed (commented):
Overall, the code is simple and straightforward, though there are some methods/variables whose name does not clearly describe their purpose in context. I have also noticed a few potential issues in the code that may affect maintainability. 

Likewise, I would suggest moving the argument parsing out of `Task` subclasses and into `Parser` to clearly separate their responsibilities.

moingshoing committed:
Set up checkstyle to detect coding style violations
Edit code to fix coding style violations

moingshoing committed:
Add Main.java, MainWindow.java, DialogBox.java, Launcher.java to add GUI to chatbot
Update Aris to support GUI

xinnnyeee commented on src/main/java/aris/Aris.java:
well documented methods!

moingshoing committed:
Add style class for add task, mark task, delete task, and find task

xinnnyeee commented on src/main/java/aris/task/Deadline.java:
```diff
@@ -0,0 +1,62 @@
+        this.deadlineString = String.format(" (by: %s)", this.deadline);
+        this.isDone = false;
+    }
+
```
interesting way to handle user input directly! perhaps would be more obvious if you name the fullDescription as userInput instead? but up to you ^_^

xinnnyeee commented on src/main/java/aris/ui/Ui.java:
```diff
@@ -0,0 +1,30 @@
+    public void greet() {
+        format("Hello! I'm Aris!\n|･ω･｀) < hi.");
+    }
+
```
i like how you used these cute text-emojis!

xinnnyeee commented on src/main/java/aris/Aris.java:
```diff
@@ -0,0 +1,125 @@
+                // Fallthrough
+            default: // Any other text
+                arisUi.format("Sorry forgot to code this bit ¯\\_(._.)_/¯");
+            }
```
is it better to let user know if their problem is incomplete command or wrong command?

xinnnyeee reviewed (commented):
Generally, LGTM, a very cute program which i like! you shd look at my program too HHAAHAH.

moingshoing committed:
MainWindow.java: Add gif file as background

moingshoing committed:
Aris.java: Update execute method to respond to "bye" command
Change code according to checkstyle coding standards

moingshoing committed:
Parser.java: Add if statement to parse greeting commands
Command.java: Add command GREET
Aris.java: Add response to greeting commands

moingshoing committed:
Aris.java: Add assertion to prevent empty reply messages

Added an assertion to check that the reply message is not empty, preventing unexpected behavior.
This ensures proper response formatting and improves robustness.

moingshoing committed:
TaskList.java: Add assertions to ensure data integrity

Added assertions to verify that the task list, tasks, and indices are valid before performing operations.
This helps catch logical errors early and ensures that methods operate on valid data.

Assertions added for:
* Ensuring `list` is not null before operations.
* Preventing `null` tasks from being added or modified.

moingshoing committed:
TaskList.java: Replace magic strings with constants

Replaced hardcoded messages for "Number is out of range" and "No results found" with constants.
This improves maintainability, reduces duplication, and enhances readability.

Constants added:
* `NUMBER_OUT_OF_RANGE`
* `NO_RESULTS_FOUND`

moingshoing committed:
Aris.java: Improve code quality and maintainability

* Replaced magic strings with constants for better readability.
* Extracted task modification, creation, and exit handling into separate methods.
* Used enhanced switch syntax for better structure.
* Improved encapsulation by making variables and methods private where applicable.
* Refactored storage saving into a dedicated method.

This improves maintainability, ensures better separation of concerns, and enhances code clarity.

moingshoing committed:
Aris.java: Remove import of Scanner

moingshoing committed:
Merge pull request #1 from moingshoing/branch-A-Assertions

A-Assertions

moingshoing committed:
Merge branch 'master' into branch-A-CodeQuality

moingshoing committed:
Merge pull request #2 from moingshoing/branch-A-CodeQuality

A-CodeQuality

moingshoing committed:
Update code for readability

moingshoing committed:
Add Help window for in-app guidance

* Implemented help.fxml to display formatted command instructions.
* Created HelpController.java to manage Help page behavior.

moingshoing committed:
Add classes to show help window when prompted

moingshoing committed:
Replace HelpController with HelpWindow

The previous HelpController implementation and help.fxml were buggy, causing
XML parsing errors and failing to load properly. To fix this:

* Removed `HelpController.java` and `help.fxml`.
* Introduced `HelpWindow.java` as the new Help window controller.
* Created `HelpWindow.fxml` with a structured and maintainable UI.
* Updated `Aris.java` to integrate the new Help window system.

moingshoing committed:
Improve Help Window UI for better readability

* Enhanced text alignment and spacing for clearer presentation.
* Improved command formatting to make the Help guide easier to read.
* Ensured better user experience with a cleaner and more structured layout.

moingshoing committed:
Merge branch 'branch-BCD-Extension'

# Conflicts:
#	src/main/resources/view/DialogBox.fxml
#	src/main/resources/view/MainWindow.fxml

moingshoing committed:
Change version of JavaFx to 17.0.7

moingshoing committed:
MainWindow.java: Add dialog box to prompt user to type 'help'

moingshoing committed:
Ui.java: Remove dead code from the codebase

moingshoing committed:
Remove JavaDoc comment

moingshoing committed:
Aris.java: Open YouTube video when typing 'punch' command

* Added `sendPunch()` method in `Aris.java` to open a YouTube video.
* Integrated JavaFX `HostServices` to launch the video in the
default browser.
* Ensured smooth execution when the user types "punch" in the chat.

This makes the 'punch' command more interactive by directing users
to an external video.

moingshoing committed:
Replace 'punch' command with 'suisei' command to open
random YouTube videos

* Removed 'punch' command and its associated functionality.
* Added 'suisei' command in `Aris.java` that selects a random
YouTube link from a predefined array.
* Integrated JavaFX `HostServices` to open the selected video
in the default browser.

moingshoing committed:
HelpWindow.java: Update Help Window text

moingshoing committed:
Main.java: Add title to main window

moingshoing committed:
Edit greet message

moingshoing committed:
Add Ui.png

moingshoing committed:
Update README.md

moingshoing committed:
Update README

moingshoing committed:
Reformat table in README

moingshoing committed:
Update README.md
